### ISSUE TITLE
When coockie.encryption=true, NPE in CookieParser.decrypt(CookieParser.java:134) is thrown

### ISSUE THREAD

**Issue by ismagilov** [svenkubiak/mangooio#284]
I tried to turn on cookie encryption, and got a NPE. Before testing I had cleared all cookies for `localhost` domain.

Controller method:
```
    public Response todo() {
        return Response.withOk().andJsonBody("TODO");
    }
```

```
    cookie:
        name       : todobackend-SESSION
        expires    : 86400
        encryption : false
```

<img width="983" alt="screen shot 2017-04-07 at 17 22 53" src="https://cloud.githubusercontent.com/assets/6725061/24796462/3f6813ac-1bb7-11e7-98ac-f1619e1cca53.png">

```
    cookie:
        name       : todobackend-SESSION
        expires    : 86400
        encryption : true
```

<img width="932" alt="screen shot 2017-04-07 at 17 23 48" src="https://cloud.githubusercontent.com/assets/6725061/24796472/4aaa2cdc-1bb7-11e7-966b-694cdd9673f4.png">

Stacktrace:
```
07.04.2017 16:23:41,850 [XNIO-1 I/O-7] ERROR io.mangoo.routing.handlers.ExceptionHandler - Internal server exception
java.lang.NullPointerException: null
	at io.mangoo.utils.cookie.CookieParser.decrypt(CookieParser.java:134) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.utils.cookie.CookieParser.hasValidSessionCookie(CookieParser.java:66) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.InboundCookiesHandler.getSessionCookie(InboundCookiesHandler.java:69) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.InboundCookiesHandler.handleRequest(InboundCookiesHandler.java:46) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.LocaleHandler.nextHandler(LocaleHandler.java:60) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.LocaleHandler.handleRequest(LocaleHandler.java:49) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.LimitHandler.nextHandler(LimitHandler.java:92) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.LimitHandler.handleRequest(LimitHandler.java:47) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.DispatcherHandler.nextHandler(DispatcherHandler.java:190) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.DispatcherHandler.handleRequest(DispatcherHandler.java:162) ~[mangooio-core-4.3.0.jar:?]
	at io.undertow.server.RoutingHandler.handleRequest(RoutingHandler.java:93) ~[undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.handlers.PathHandler.handleRequest(PathHandler.java:94) ~[undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.handlers.ExceptionHandler.handleRequest(ExceptionHandler.java:29) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:211) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpReadListener.handleEventWithNoRunningRequest(HttpReadListener.java:243) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpReadListener.handleEvent(HttpReadListener.java:134) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:148) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:92) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:51) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:291) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:286) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.QueuedNioTcpServer$1.run(QueuedNioTcpServer.java:128) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:580) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:464) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
07.04.2017 16:23:41,855 [XNIO-1 I/O-7] ERROR io.mangoo.routing.handlers.ExceptionHandler - Failed to pass an exception to the frontend
java.lang.NullPointerException: null
	at io.mangoo.templating.freemarker.TemplateEngineFreemarker.renderException(TemplateEngineFreemarker.java:130) ~[mangooio-core-4.3.0.jar:?]
	at io.mangoo.routing.handlers.ExceptionHandler.handleRequest(ExceptionHandler.java:47) [mangooio-core-4.3.0.jar:?]
	at io.undertow.server.handlers.ExceptionHandler.handleRequest(ExceptionHandler.java:34) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:211) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpReadListener.handleEventWithNoRunningRequest(HttpReadListener.java:243) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpReadListener.handleEvent(HttpReadListener.java:134) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:148) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:92) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:51) [undertow-core-1.4.11.Final.jar:1.4.11.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:291) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:286) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.QueuedNioTcpServer$1.run(QueuedNioTcpServer.java:128) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:580) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:464) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
```

**Comment by svenkubiak** [comment]
Fixed. Thanks for reporting.

**Comment by svenkubiak** [comment]
Fixed in 4.3.1
